packages/grafana-ui/src/components/Dropdown/ButtonSelect.tsx

```diff
@@ -89,6 +89,9 @@ const ButtonSelectComponent = <T,>(props: Props<T>) => {
 
 ButtonSelectComponent.displayName = 'ButtonSelect';
 
+// needed to properly forward the generic type through React.memo
+// see https://github.com/DefinitelyTyped/DefinitelyTyped/issues/37087#issuecomment-656596623
+// eslint-disable-next-line @typescript-eslint/consistent-type-assertions
 export const ButtonSelect = React.memo(ButtonSelectComponent) as typeof ButtonSelectComponent;
```

Comment:
Does this work?

```suggestion
export const ButtonSelect = React.memo<ButtonSelectComponent>(ButtonSelectComponent);
```

Reply:
no 😢

Reply:
hmm, if you do `React.memo<typeof ButtonSelectComponent>(ButtonSelectComponent);`?